src/Microsoft.AspNet.Mvc.Core/ControllerActionExecutor.cs

```diff
@@ -172,7 +172,7 @@ private static void ThrowIfWrappedTaskInstance(Type actualTypeReturned, string m
         /// </summary>
         private static async Task<object> CastToObject(Task task)
         {
-            await task;
+            await task.ConfigureAwait(false);
             return null;
```

Comment:
This code is a bit exotic; was going to do something different; but probably needs a double check instead?